packages/studio-be/src/studio/nlu/nlu-client/cloud/client.ts

```diff
@@ -40,6 +40,12 @@ export class CloudClient extends StanClient {
 
     this.axios.interceptors.request.use(this._requestInterceptor(tokenCache).bind(this) as any)
     this.axios.interceptors.response.use(undefined, this._errorInterceptor(this.axios as any, tokenCache).bind(this))
+
+    this.modelWeights.axios.interceptors.request.use(this._requestInterceptor(tokenCache).bind(this) as any)
+    this.modelWeights.axios.interceptors.response.use(
+      undefined,
+      this._errorInterceptor(this.axios as any, tokenCache).bind(this)
+    )
```

Comment:
Since the `ModelTransferClient` does not use the same `axios` instance as the `NLUClient`, we have to set the interceptors twice (another option would be to refactor the NLU client so clients share the same axios instance). This does not really impact performances in any way since the interceptor's handlers use the same cache.

Reply:
this is fine